frontend/src/infra/MyProvider.js

```diff
@@ -40,40 +29,48 @@ class MyProvider extends Component {
     });
     const req = { creatorID: playerId };
     axios
-      .post(SERVER_ADDRESS + ":" + SERVER_PORT + "/room/create", req)
-      .then(res => {
-        const { roomID, a, creationDate } = res.data;
+      .post(`${SERVER_ADDRESS}:${SERVER_PORT}/room/create`, req)
+      .then((res) => {
+        console.log(res.data);
+        const { roomID, a, creationDate, socketPort } = res.data;
+        createSocketAndListen(socketPort, "NewPlayer", (data) => {
+          this.setState({
+            playerList: data.roomPlayers,
+          });
+        });
         this.setState({
           currentRoom: roomID,
-          roomCreationDate: creationDate
+          roomCreationDate: creationDate,
         });
       });
   };
-
-  joinGame = roomId => {
+  joinGame = (roomId) => {
     const playerId = Shortid.generate();
     this.setState({
       phase: PHASE.WAITING_ROOM,
       currentRoom: roomId,
-      playerId: playerId
+      playerId: playerId,
     });
   };
 
   onNameEntered = (name) => {
     axios
       .get(
-        SERVER_ADDRESS +
-          ":" +
-          SERVER_PORT +
-          "/room/join?userID=" +
-          this.state.playerId +
-          "&roomID=" +
-          this.state.currentRoom +
-          "&playerName=" +
-          name
+        `${SERVER_ADDRESS}:${SERVER_PORT}/room/join?userID=${this.state.playerId}&roomID=${this.state.currentRoom}`
```

Comment:
room should be a const

Reply:
https://github.com/aviadch/Mafia/commit/8df6d0bac164de4c37e949f7d2434af26f5ed94c